Source/Core/VideoBackends/D3D/D3DState.h

```diff
@@ -224,43 +225,27 @@ class StateManager
 private:
   enum DirtyFlags
   {
-    DirtyFlag_Texture0 = 1 << 0,
-    DirtyFlag_Texture1 = 1 << 1,
-    DirtyFlag_Texture2 = 1 << 2,
-    DirtyFlag_Texture3 = 1 << 3,
-    DirtyFlag_Texture4 = 1 << 4,
-    DirtyFlag_Texture5 = 1 << 5,
-    DirtyFlag_Texture6 = 1 << 6,
-    DirtyFlag_Texture7 = 1 << 7,
-
-    DirtyFlag_Sampler0 = 1 << 8,
-    DirtyFlag_Sampler1 = 1 << 9,
-    DirtyFlag_Sampler2 = 1 << 10,
-    DirtyFlag_Sampler3 = 1 << 11,
-    DirtyFlag_Sampler4 = 1 << 12,
-    DirtyFlag_Sampler5 = 1 << 13,
-    DirtyFlag_Sampler6 = 1 << 14,
-    DirtyFlag_Sampler7 = 1 << 15,
-
-    DirtyFlag_PixelConstants = 1 << 16,
-    DirtyFlag_VertexConstants = 1 << 17,
-    DirtyFlag_GeometryConstants = 1 << 18,
-
-    DirtyFlag_VertexBuffer = 1 << 19,
-    DirtyFlag_IndexBuffer = 1 << 20,
-
-    DirtyFlag_PixelShader = 1 << 21,
-    DirtyFlag_VertexShader = 1 << 22,
-    DirtyFlag_GeometryShader = 1 << 23,
-
-    DirtyFlag_InputAssembler = 1 << 24,
-    DirtyFlag_BlendState = 1 << 25,
-    DirtyFlag_DepthState = 1 << 26,
-    DirtyFlag_RasterizerState = 1 << 27,
-    DirtyFlag_Framebuffer = 1 << 28
+    DirtyFlag_Texture0 = 0,
+    DirtyFlag_Sampler0 = DirtyFlag_Texture0 + VideoCommon::MAX_PIXEL_SHADER_SAMPLERS,
+    DirtyFlag_PixelConstants = DirtyFlag_Sampler0 + VideoCommon::MAX_PIXEL_SHADER_SAMPLERS,
+    DirtyFlag_VertexConstants,
+    DirtyFlag_GeometryConstants,
+
+    DirtyFlag_VertexBuffer,
+    DirtyFlag_IndexBuffer,
+
+    DirtyFlag_PixelShader,
+    DirtyFlag_VertexShader,
+    DirtyFlag_GeometryShader,
+
+    DirtyFlag_InputAssembler,
+    DirtyFlag_BlendState,
+    DirtyFlag_DepthState,
+    DirtyFlag_RasterizerState,
+    DirtyFlag_Framebuffer
   };
 
-  u32 m_dirtyFlags = ~0u;
+  std::bitset<DirtyFlags::DirtyFlag_Framebuffer + 1> m_dirtyFlags;
```

Comment:
Suggestion: Add a DirtyFlag_Max as the last member of the enum to use here? Fairly common pattern.

Reply:
Ah, I'm aware of that pattern but didn't think to use it here.  Thanks.  Updated.